packages/store-sync/src/dozer/fetchRecordsSql.ts

```diff
@@ -0,0 +1,69 @@
+import { DecodeDozerRecordsResult, DozerQueryResult, decodeDozerRecords } from "./decodeDozerRecords";
+import { Hex } from "viem";
+import { TableQuery } from "./common";
+import { Table } from "@latticexyz/config";
+
+type DozerResponseSuccess = {
+  block_height: string;
+  result: DozerQueryResult[];
+};
+
+type DozerResponseFail = { msg: string };
+
+type DozerResponse = DozerResponseSuccess | DozerResponseFail;
+
+function isDozerResponseFail(response: DozerResponse): response is DozerResponseFail {
+  return "msg" in response;
+}
+
+type FetchRecordsSqlArgs = {
+  dozerUrl: string;
+  storeAddress: Hex;
+  queries: TableQuery[];
+};
+
+type FetchRecordsSqlResult =
+  | {
+      blockHeight: bigint;
+      result: {
+        table: Table;
+        records: DecodeDozerRecordsResult;
+      }[];
+    }
+  | undefined;
```

Comment:
not blocking but fwiw I found it quite nice to model this sort of thing with arktype, then can parse/validate/return strongly typed response from a JSON API request